mantidproject/mantid
# Engineering Diffraction GUI enable loading and plotting runs in the new GSAS tab

The new GSAS tab in the ED GUI was designed and hooked up to the main GUI (as more parameters are added to `GSASIIRefineFitPeaks` this will be augmented, but it is adequate for now). Enough functionality was added to the GUI to load runs and plot them in the plot area.

Hopefully this will be the last of the ED GUI PRs with really big diffs, as adding further features should now just be a case of filling in methods.

**To test:**

Sample data files [here](https://github.com/mantidproject/mantid/files/1652777/engggui_pr_data.zip). 

Load up the ED GUI (`Interfaces -> Diffraction -> Engineering Diffraction`) then click `GSAS Fitting` tab.

Click the first **Browse** button to browse for focused run data (in the zip file). Load all 3 runs, and make sure `277208_1`, `277208_2` and `277209_2` are displayed in the list widget next to the plot. Clicking each of them in turn should plot the run. Try only loading 2 runs, and make sure the third is added correctly once loaded on its own (you'll need to exit the GUI to get rid of the loaded runs).

Make sure the zoom tool works nicely as well - zoom by clicking and dragging, and reset by right-clicking

Partially address #15552 

**Release Notes** 
Will be added to the release notes once the user can do refinement from the GUI

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
